text/0524-vault-standard.md

```diff
@@ -0,0 +1,949 @@
+
+  - **`OP_VAULT_NOTIFICATION`** <a id="op_vault_notification"></a>: For withdrawing or refunding TON.
+
+    | Field                     | Type                                                | Description              |
+    | ------------------------- | --------------------------------------------------- | ------------------------ |
+    | `OP_VAULT_NOTIFICATION`   | [Opcode](#opcode)                                   | `0x86eba146`             |
+    | `queryId`                 | [QueryId](#queryid)                                 | Unique query identifier. |
+    | `vaultNotificationParams` | [VaultNotificationParams](#vaultnotificationparams) |                          |
+
+  - **`OP_VAULT_NOTIFICATION_FP`** <a id="op_vault_notification_fp"></a>: For minting shares, withdrawing, or refunding Jetton.
+
+    | Field                      | Type                                                | Description  |
+    | -------------------------- | --------------------------------------------------- | ------------ |
+    | `OP_VAULT_NOTIFICATION_FP` | [Opcode](#opcode)                                   | `0xb00d7656` |
+    | `vaultNotificationParams`  | [VaultNotificationParams](#vaultnotificationparams) |              |
+
+  - **`OP_VAULT_NOTIFICATION_EC`** <a id="op_vault_notification_ec"></a>: For withdrawing or refunding Extra Currency.
+
+    | Field                      | Type                                                | Description  |
+    | -------------------------- | --------------------------------------------------- | ------------ |
+    | `OP_VAULT_NOTIFICATION_EC` | [Opcode](#opcode)                                   | `0x1f9e644b` |
+    | `vaultNotificationParams`  | [VaultNotificationParams](#vaultnotificationparams) |              |
+  - **`TL-B for Options, Callbacks, and Notifications`**
+    ```
+    vault_options$_ 
+      custom_data:(Maybe ^Cell) = VaultOptions;
+
+    callback_params$_
+      include_body:Bool
+      payload:^Cell = CallbackParams;
+
+    callbacks$_
+      success_callback:(Maybe ^CallbackParams)
+      failure_callback:(Maybe ^CallbackParams) = Callbacks;
+
+    vault_notification_params$_
+      result_code:ResultCode
+      initiator:MsgAddress
+      callback_payload:(Maybe ^Cell)
+      in_body:(Maybe ^Cell) = VaultNotificationParams;
+
+    vault_notification#86eba146
+      query_id:QueryId
+      vault_notification_params:VaultNotificationParams = InternalMsgBody;
+
+    vault_notification_fp#b00d7656
+      vault_notification_params:VaultNotificationParams = ForwardPayload;
+
+    vault_notification_ec#1f9e644b
+      vault_notification_params:VaultNotificationParams = InternalMsgBody;
+    ```
+
+**Deposit (For TON)**
+
+![deposit-ton](../assets/0524-vault-standard/deposit-ton.png)
+
+- **Description**: Mint shares to `receiver` by depositing exactly `depositAmount` of TON.
```

Comment:
it would be great to specify TL-B for minting shares. TEP-74 doesn't specify it but most of the implementations are using the same approach - internal_transfer message sent from master to jetton wallet address:

```
internal_transfer  query_id:uint64 amount:(VarUInteger 16) from:MsgAddress
                     response_address:MsgAddress
                     forward_ton_amount:(VarUInteger 16)
                     forward_payload:(Either Cell ^Cell)
                     = InternalMsgBody;
```

Having the same format for all mints streamlines indexers/explorers integration